modules/core/src/smithy4s/LocalDate.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright 2021-2025 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+import scala.util.control.{NoStackTrace, NonFatal}
+
+case class LocalDate private (epochDay: Long)
+    extends LocalDatePlatform {
+
+  def isAfter(other: LocalDate): Boolean = {
+    val diff = epochDay - other.epochDay
+    diff > 0 
+  }
+
+  override def toString: String = {
+    val s = new java.lang.StringBuilder(32)
+    var marchZeroDay =
+      epochDay + 719468 // 719468 == 719528 - 60 == days 0000 to 1970 - days 1st Jan to 1st Mar
+    var adjustYear = 0
+    if (marchZeroDay < 0) { // adjust negative years to positive for calculation
+      val adjust400YearCycles = TimeUtil.to400YearCycle(marchZeroDay + 1) - 1
+      adjustYear = adjust400YearCycles * 400
+      marchZeroDay -= adjust400YearCycles * 146097L
+    }
+    var year = TimeUtil.to400YearCycle(marchZeroDay * 400 + 591)
+    var marchDayOfYear = TimeUtil.toMarchDayOfYear(marchZeroDay, year)
+    if (marchDayOfYear < 0) { // fix year estimate
+      year -= 1
+      marchDayOfYear = TimeUtil.toMarchDayOfYear(marchZeroDay, year)
+    }
+    val marchMonth =
+      (marchDayOfYear * 17135 + 6854) >> 19 // (marchDayOfYear * 5 + 2) / 153
+    year += (marchMonth * 3277 >> 15) + adjustYear // year += marchMonth / 10 + adjustYear (reset any negative year and convert march-based values back to january-based)
+    val month = marchMonth +
+      (if (marchMonth < 10) 3
+       else -9)
+    val day =
+      marchDayOfYear - ((marchMonth * 1002762 - 16383) >> 15) // marchDayOfYear - (marchMonth * 306 + 5) / 10 + 1
+
+    TimeUtil.append4Digits(year, s)
+    TimeUtil.append2Digits(month, s.append('-'))
+    TimeUtil.append2Digits(day, s.append('-'))
+    s.toString
+  }
+}
+
+object LocalDate extends LocalDateCompanionPlatform {
+
+  val epoch = LocalDate(0)
+
+  def apply(epochDay: Long): LocalDate = new LocalDate(epochDay)
+
+  def apply(
+      year: Int, 
+      month: Int,
+      day: Int,
+  ): LocalDate = {
```

Comment:
What is the reason for throwing exceptions with invalid input rather than using an Either or Option? Perhaps we should make it clear this method is unsafe if we keep as is `unsafeApply` or something.

Reply:
Just saw how it was done in the `Timestamp` and decided to keep it consistent with that, but I have nothing against making this return an `Either` or `Option`

 @kubukoz @Baccata have any opinions on this?

Reply:
Okay yeah I think keeping it consistent is fine and we can refactor all of them in one go if we decide that's the right path forward